File: 2022-2023/sourcecode/lab/lab5/ViewPagerFragmentStateAdapterJava/app/src/main/java/pl/edu/uwr/pum/viewpagerfragmentstateadapterjava/PagerAdapter.java
package pl.edu.uwr.pum.viewpagerfragmentstateadapterjava;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class PagerAdapter extends FragmentStateAdapter {


    public PagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return new TemplateFragment(position);
    }

    @Override
    public int getItemCount() {
        return MainActivity.num;
    }
}

File: 2022-2023/sourcecode/lab/lab5/ViewPagerFragmentStateAdapterKotlin/app/src/main/java/pl/edu/uwr/pum/viewpagerfragmentstateadapterkotlin/PagerAdapter.kt
package pl.edu.uwr.pum.viewpagerfragmentstateadapterkotlin

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class PagerAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {

    override fun getItemCount(): Int = MainActivity.num

    override fun createFragment(position: Int): Fragment {
        return TemplateFragment(position)
    }
}